vm/interpreter/src/default_runtime.rs

```diff
@@ -985,26 +985,35 @@ where
     ) -> Result<HashMap<Address, Vec<bool>>, Box<dyn StdError>> {
         // Gas charged for batch verify in actor
 
-        let out = vis
-            .par_iter()
-            .map(|(&addr, seals)| {
-                let results = seals
-                    .par_iter()
-                    .map(|s| {
-                        if let Err(err) = V::verify_seal(s) {
-                            debug!(
-                                "seal verify in batch failed (miner: {}) (err: {})",
-                                addr, err
-                            );
-                            false
-                        } else {
-                            true
-                        }
-                    })
-                    .collect();
-                (addr, results)
-            })
-            .collect();
+        let cpus = num_cpus::get();
+        // Create a local threadpool just for running verify seal in this epoch
+        let batch_pool = rayon::ThreadPoolBuilder::new()
+            .num_threads(cpus)
+            .thread_name(|id| format!("batch_verify_seal_{}", id))
+            .build()?;
+
+        let out = batch_pool.scope(|_scope| {
+            vis.par_iter()
+                .with_min_len(cpus)
```

Comment:
Should this be `min_len` or `max_len` ?

Reply:
`min_len` sets a floor on the chunk size which avoids scheduling too many blocking tasks across the threads in this blocking thread pool. Max would set a ceiling on the chunk size, which would have the opposite effect. We determined that number of chunks in the outer loop and inner loop is the source of the stack overflow.

Reply:
Mhm, thats a better heuristic, actually.

Reply:
I see, thanks for elaborating on that. Do we then not intend to use a bound on the inner `par_iter` as well?

Reply:
No. The average len of the inner par_iter is 1.